### DISCUSSION TITLE
Disk layout

### DISCUSSION THREAD

**Discussion by bambooCZ** [kwankiu/archlinux-installer-rock5#18] [General]
Hello,

I am just wondering if it is really necessary to have the empty 16 MB FAT partition and then the second large root partition with type EFI formatted with ext4 with size in gigabytes...

**Comment by kwankiu** [comment]
Ah, the 16MB empty partition is Radxa’s image’s approach, you can put a config.txt there like the raspberry pi 

since rock 5B’s u-boot is in the on board SPI and it is able to boot from the ext4 partition,I think we should be able to start ext4 at 16MiB instead of the current 32MiB, u can run the script to build images with different partitions sizes and test. Just look at the GitHub workflow files for how to run the script.

**Comment by bambooCZ** [comment]
Okay now I've played with it and I see that it isn't possible without GPT type 1 - EFI System Partition. Otherwise it won't boot...

So if I want to have calm sleep which is prevented by using EFI partition as root partition (facepalm - would be interested who is to blame wheater uboot or radaxa) then I can separate boot partition and figure out how to tell uboot to look for kernel in EFI/Arch/<kernel> or just run of EFI System partition (facepalm again). This is against all RFCs all standards all intuition.
